lib/resty/healthcheck.lua

```diff
@@ -1475,6 +1505,29 @@ function _M.new(opts)
 
         local cur_time = ngx_now()
         for _, checker_obj in ipairs(hcs) do
+          -- clear targets marked for delayed removal
+          locking_target_list(checker_obj, function(target_list)
+            local removed_targets = {}
+            for i, target in ipairs(target_list) do
+              if target.purge_time and target.purge_time <= cur_time then
+                table_insert(removed_targets, target)
+                table_remove(target_list, i)
```

Comment:
this is dangerous, you're removing from the array you're iterating, so the next item will be skipped because `i` would still advance.

either use a `while` loop and don't advance when removing, or iterate backwards, or don't remove but rather copy to another table.

Reply:
😳 🤦

I don't know what I was thinking. Thank you!